gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
We are indeed already testing `if len(self.WORKERS.keys()) < self.num_workers` . The goal of this function is to make sure that this number is always at least equal to self.num_workers. Why not moving the debug log at the end of the if block? And add it in different places where we detect when a work exit?